phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -634,6 +638,23 @@ private Void addDataColInfo(final PTable dataTable, Expression expression) {
                         }
                         return null;
                     }
+
+                    private boolean tableHasOnlyPkCols(PTable dataTable, byte[] cf) {
+                        try {
+                            dataTable.getColumnFamily(cf);
```

Comment:
why is this call necessary ?

Reply:
We could make this an API on the PTable object itself ? `HasOnlyPKCols` that might be cleaner.

Reply:
pk columns in the PTable do not have any CF associated with them, hence this call is necessary.

```
    public static boolean isPKColumn(PColumn column) {
        return column.getFamilyName() == null;
    }
```

Reply:
But why is the function `tableHasOnlyPkCols()` taking a column object as an input is something I am not able to understand

Reply:
because `dataTable.getColumnFamily(cf)` will fail with CFNFE for certain since the PTable has only pk columns but you are correct that PTable should have this utility rather than keeping it here

Reply:
@tkhurana this is taken care of in the latest revision